src/librustc_mir/transform/validate.rs

```diff
@@ -77,4 +86,94 @@ impl<'a, 'tcx> Visitor<'tcx> for TypeChecker<'a, 'tcx> {
+            }
+            TerminatorKind::SwitchInt { targets, .. } => {
+                if targets.is_empty() {
+                    self.fail(
+                        terminator.source_info.span,
+                        "encountered `SwitchInt` terminator with no target to jump to",
+                    );
+                }
+                for target in targets {
+                    self.check_bb(terminator.source_info.span, *target);
+                }
+            }
+            TerminatorKind::Drop { target, unwind, .. } => {
+                self.check_bb(terminator.source_info.span, *target);
+                if let Some(unwind) = unwind {
+                    self.check_bb(terminator.source_info.span, *unwind);
+                }
+            }
+            TerminatorKind::DropAndReplace { target, unwind, .. } => {
+                self.check_bb(terminator.source_info.span, *target);
+                if let Some(unwind) = unwind {
+                    self.check_bb(terminator.source_info.span, *unwind);
+                }
+            }
+            TerminatorKind::Call { func, destination, cleanup, .. } => {
+                let func_ty = func.ty(&self.body.local_decls, self.tcx);
+                match func_ty.kind {
+                    ty::FnPtr(..) | ty::FnDef(..) => {}
+                    _ => self.fail(
+                        terminator.source_info.span,
+                        format!("encountered non-callable type {} in `Call` terminator", func_ty),
+                    ),
+                }
+                if let Some((_, target)) = destination {
+                    self.check_bb(terminator.source_info.span, *target);
+                }
+                if let Some(cleanup) = cleanup {
+                    self.check_bb(terminator.source_info.span, *cleanup);
+                }
+            }
+            TerminatorKind::Assert { cond, target, cleanup, .. } => {
+                let cond_ty = cond.ty(&self.body.local_decls, self.tcx);
+                if cond_ty != self.tcx.types.bool {
+                    self.fail(
+                        terminator.source_info.span,
+                        format!(
+                            "encountered non-boolean condition of type {} in `Assert` terminator",
+                            cond_ty
+                        ),
+                    );
+                }
+                self.check_bb(terminator.source_info.span, *target);
+                if let Some(cleanup) = cleanup {
+                    self.check_bb(terminator.source_info.span, *cleanup);
+                }
+            }
+            TerminatorKind::Yield { resume, drop, .. } => {
+                self.check_bb(terminator.source_info.span, *resume);
+                if let Some(drop) = drop {
+                    self.check_bb(terminator.source_info.span, *drop);
+                }
+            }
+            TerminatorKind::FalseEdges { real_target, imaginary_target } => {
```

Comment:
Btw, does anyone know why this is named in plural? Shouldn't it be `FalseEdge` instead?
Cc @matthewjasper

Reply:
I think that it may have been accurate in the past. Feel free to change it.